test.js

```diff
@@ -717,4 +728,44 @@ function runTest (client, db) {
       })
     })
   })
+
+  test('subscription should expire after client disconnected', function (t) {
+    dbopts.ttl = {
+      subscriptions: 1
+    }
+    dbopts.ttlAfterDisconnected = true
+    const instance = persistence(dbopts)
+    const emitter = mqemitterMongo(dbopts)
+    const client = { id: 'client1' }
+    instance.broker = toBroker('1', emitter)
+    instance.on('ready', () => {
+      instance.addSubscriptions(client, [{
+        topic: 'hello',
+        qos: 1
+      }], function (err) {
+        t.error(err)
+        db.collection('subscriptions').findOne({ clientId: client.id, topic: 'hello' }, function (err, result) {
+          t.error(err)
+          t.notEqual(result, null, 'must return subscription')
+          t.equal(result.disconnected, undefined, 'disconnected should not be set')
+          instance.broker.emit('clientDisconnect', client)
+          setTimeout(() => {
```

Comment:
I would remove this timeout and keep only the second one, you are already checking that the sub is present after adding it